R/plotAbundanceDensity.R

```diff
@@ -0,0 +1,259 @@
+#'                      n = 10, colour_by = "nationality", log_scale_x_axis = TRUE)
+#'                      
+#' # Plots the relative abundance of 10 most abundant taxa as a density plot.
+#' # X-axis is log-scaled
+#' plotAbundanceDensity(tse, layout = "density", abund_values = "relabundance", n = 10,
+#'                      log_scale_x_axis = TRUE)
+#'                      
+#' # Plots the relative abundance of 10 most abundant taxa as a point plot.
+#' # point shape is changed from default (21) to 41.
+#' plotAbundanceDensity(tse, layout = "point", abund_values = "relabundance", n = 10,
+#'                      point_shape = 41)
+#' 
+NULL
+
+#' @rdname plotAbundanceDensity
+#' @export
+setGeneric("plotAbundanceDensity", signature = c("object"),
+           function(object,
+                    layout = c("jitter", "density", "point"),
+                    abund_values = "counts",
+                    n = 25,
+                    colour_by = NULL, 
+                    ...)
+             standardGeneric("plotAbundanceDensity"))
+
+#' @rdname plotAbundanceDensity
+#' @export
+setMethod("plotAbundanceDensity", signature = c(object = "SummarizedExperiment"),
+    function(object,
+             layout = c("jitter", "density", "point"),
+             abund_values = "counts",
+             n = 25, 
+             colour_by = NULL, 
+             ...){
+        ############################# Input Check ##############################
+        # Check layout
+        layout <- match.arg(layout, several.ok = FALSE)
+        # Checks abund_values
+        .check_assay_present(abund_values, object)
+        # Checks n
+        if( !(length(n)==1 && is.numeric(n) && n%%1==0 && n>0) ){
+            stop("'n' must be a positive integer.", call. = FALSE)
+        }
+        # Checks colour_by
+        if( !is.null(colour_by) &&
+            (!.is_a_string(colour_by) ||
+            !(colour_by %in% names(colData(object)))) ){
+            stop("'colour_by' must be a name of column of colData(object)",
+                 call. = FALSE)
+        }
+        ########################### Input Check end ############################
+        # Gets data that will be plotted. Gets a list
+        density_data_list <- .incorporate_density_data(object = object,
+                                                       abund_values = abund_values,
+                                                       n = n,
+                                                       colour_by = colour_by)
+        # Extracts the density data and aesthetic from the list
+        density_data <- density_data_list$density_data
+        colour_by <- density_data_list$colour_by
+        
+        # Gets the plot from plotter
+        plot_out <- .density_plotter(density_data = density_data, 
+                                     layout = layout,
+                                     xlab = abund_values,
+                                     ylab = "Taxa",
```

Comment:
This is a hard coded value and it cannot be changed by the user. I think this should, if anything an argument with a default, but not hard coded.

Reply:
Changed